crypto/ed25519/ed25519.go

```diff
@@ -0,0 +1,253 @@
+package ed25519
+
+import (
+	"bytes"
+	"crypto/subtle"
+	"fmt"
+
+	"github.com/tendermint/ed25519"
+	"github.com/tendermint/ed25519/extra25519"
+	amino "github.com/tendermint/go-amino"
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	cmn "github.com/tendermint/tendermint/libs/common"
+)
+
+//-------------------------------------
+
+var _ crypto.PrivKey = PrivKeyEd25519{}
+
+const (
+	Ed25519PrivKeyAminoRoute   = "tendermint/PrivKeyEd25519"
+	Ed25519PubKeyAminoRoute    = "tendermint/PubKeyEd25519"
+	Ed25519SignatureAminoRoute = "tendermint/SignatureEd25519"
+)
+
+var cdc = amino.NewCodec()
+
+func init() {
+	// NOTE: It's important that there be no conflicts here,
+	// as that would change the canonical representations,
+	// and therefore change the address.
+	// TODO: Add feature to go-amino to ensure that there
+	// are no conflicts.
```

Comment:
This are not new TODOs, right? It's a bit odd that this shows up as new file instead of showing changes to the old one.

Reply:
It's showing that because the file was moved + edited for comments. It is an old to do

Reply:
Thanks for the clarification.